std/testing/asserts_test.ts

```diff
@@ -245,6 +246,33 @@ test("testingAssertFailWithWrongErrorClass", function (): void {
   );
 });
 
+test("testingAssertFailWithReturnType", function (): void {
+  assertThrows(() => {
+    throw new Error();
+    return "a string";
+  });
+});
+
+test("testingAssertFailAsync", function (): void {
+  assertThrowsAsync(
+    () => Promise.resolve(fail()),
+    AssertionError,
+    "Failed assertion."
+  );
+  assertThrowsAsync(
+    () => Promise.resolve(fail("foo")),
+    AssertionError,
+    "Failed assertion: foo"
+  );
+});
+
+test("testingAssertFailAsyncWithReturnType", function (): void {
```

Comment:
Despite the usage elsewhere in this module, please use arrow syntax for the anonymous function here.